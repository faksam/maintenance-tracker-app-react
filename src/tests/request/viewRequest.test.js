import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedViewRequest, { ViewRequest } from '../../app/components/request/ViewRequest';

const props = {
  fullname: 'fakunlesamuel@gmail.com',
  email: 'fakunlesamuel@gmail.com',
  password: 'Lifeisarace',
  confirmPassword: 'Lifeisarace',
  handleSignIn: jest.fn(),
  handleSignUp: jest.fn(),
  handleChange: jest.fn(),
  viewClickedRequest: jest.fn(),
  handleCreateRequest: jest.fn(),
  hideModalRequest: jest.fn(),
  signoutUserAction: jest.fn(),
  loadRequestsAction: jest.fn(),
  request: [{}],
  history: {},
  user: {
    username: 'joeeasy',
    email: 'joeeasy@gmail.com',
    role: 'Admin',
    id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
    iat: 1540985284931
  },
};
const initialState = {
  auth: {
    isAuthenticated: true,
    error: {},
    user: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      role: 'Admin',
      id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
      iat: 1540985284931
    }
  },
  requestReducer: {
    requests: []
  }

};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('SignInTabForm Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<ConnectedViewRequest store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<ConnectedViewRequest store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('should render without throwing an error', () => {
    const wrapper = shallow(<ViewRequest store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should a new comment', () => {
    const wrapper = shallow(<ViewRequest store={store} {...props} />);
    const instance = wrapper.instance();
    instance.handleCloseError();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<ViewRequest store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
});
