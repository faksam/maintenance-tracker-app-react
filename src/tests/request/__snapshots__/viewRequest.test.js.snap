// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInTabForm Component should render without throwing an error 1`] = `
<ViewRequest
  auth={true}
  confirmPassword="Lifeisarace"
  email="fakunlesamuel@gmail.com"
  error={Object {}}
  fullname="fakunlesamuel@gmail.com"
  handleChange={[MockFunction]}
  handleCreateRequest={[MockFunction]}
  handleSignIn={[MockFunction]}
  handleSignUp={[MockFunction]}
  hideModalRequest={[MockFunction]}
  history={Object {}}
  loadRequestsAction={[MockFunction]}
  password="Lifeisarace"
  request={
    Array [
      Object {},
    ]
  }
  signoutUserAction={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "joeeasy@gmail.com",
      "iat": 1540985284931,
      "id": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      "role": "Admin",
      "username": "joeeasy",
    }
  }
  viewClickedRequest={[MockFunction]}
/>
`;

exports[`SignInTabForm Component should render without throwing an error 2`] = `
<div
  className="modal"
  id="viewRequestModal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <span
        className="close"
        id="viewRequestSpan"
        onClick={[MockFunction]}
      >
        Ã—
      </span>
    </div>
    <div
      className="modal-body"
    >
      <div
        className="row left"
      >
        <div
          className="container"
        >
          <div
            className="col col-1"
          >
            <div
              className="row overflow"
            >
              <h2
                className="center"
              >
                <b>
                  Request
                </b>
              </h2>
              <hr
                className="style14"
              />
              <h2>
                <b>
                  Title: 
                </b>
              </h2>
              <h3
                id="viewRequestTitle"
              />
              <h2>
                <b>
                  Description:
                </b>
              </h2>
              <h3
                id="viewRequestDescription"
              />
              <h2>
                <b>
                  Date:
                </b>
              </h2>
              <h3
                id="viewRequestDate"
              >
                Nov 1, 2018
              </h3>
              <h2>
                <b>
                  Status:
                </b>
              </h2>
              <h3
                id="viewRequestStatus"
              />
            </div>
          </div>
          <br />
          <br />
          <br />
        </div>
      </div>
    </div>
    <br />
    <br />
  </div>
</div>
`;
